sdk/tests/batch.test.ts

```diff
@@ -0,0 +1,126 @@
+import 'mocha'
+import {
+  BatchError,
+  batchTxs,
+  initGearApi,
+  subscribeToNewHeads,
+  toPercentage
+} from '../src/utils.js'
+import { GearKeyring, HexString } from '@gear-js/api'
+import { Network } from '../src/consts'
+import { Invariant } from '../src/invariant'
+import { assert } from 'chai'
+import { FungibleToken } from '../src/erc20.js'
+
+const api = await initGearApi({ providerAddress: Network.Local })
+const admin = await GearKeyring.fromSuri('//Alice')
+
+let unsub: Promise<VoidFunction> | null = null
+const GRC20: FungibleToken = await FungibleToken.load(api)
+GRC20.setAdmin(admin)
+let token0Address: HexString = null as any
+let token1Address: HexString = null as any
+let invariant: Invariant = null as any
+
+describe('events', async function () {
+  this.timeout(80000)
+
+  this.beforeAll(async function () {
+    unsub = subscribeToNewHeads(api)
+  })
+  this.beforeEach(async function () {
+    this.timeout(80000)
+
+    invariant = await Invariant.deploy(api, admin, toPercentage(1n, 2n))
+    token0Address = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 0n)
+    token1Address = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 0n)
+  })
+
+  it('batch deposit and withdraw', async function () {
+    this.timeout(20000)
+    await batchTxs(api, admin, [
+      await GRC20.mintTx(admin.addressRaw, 1000000000000n, token0Address),
```

Comment:
Is the use of `await` inside `batchTxs` required?

Reply:
Yes, setting the gas limit requires using await